aiida/common/datastructures.py

```diff
@@ -80,7 +80,8 @@ class CalcInfo(DefaultFieldsAttributeDict):
         either, for example, because they contain proprietary information or because they are big and their content is
         already indirectly present in the repository through one of the data nodes passed as input to the calculation.
     * codes_info: a list of dictionaries used to pass the info of the execution of a code
-    * codes_run_mode: a string used to specify the order in which multi codes can be executed
+    * codes_run_mode: `CodeRunMode.SERIAL` or `CodeRunMode.PARALLEL` to specify the order
+        in which multi codes can be executed
```

Comment:
So if I understand correctly this keyword controls if the codes provided will be run one after the other or all at the same time, right? I don't think I would understand this as "order" in which the codes can be executed. I think the noun "mode" you use for keyword is also a better descriptor for its behavior. So something like:

```suggestion
    * codes_run_mode: the mode of execution in which the codes will be run (`CodeRunMode.SERIAL` by default, but can also be `CodeRunMode.PARALLEL`)
```
---
**SIDENOTE** (meaning, this is not super relevant, if it even makes sense we can open a separate issue for this)

I am now wondering why do we need the `CodeRunMode` construct. Are we anticipating there might be other customizable options we might pass through this keyword? Because in principle there is nothing conceptually besides serial and parallel, so this could simply be a flag no? @sphuber ?

Reply:
In a sense yes it is a "flag" as it only has two values, but the "flag" concept only really applies to a command-line interface where either you specify it or not. Even there, if you don't specify it, there is still some default value, namely the feature to which the flag corresponds being turned off. Here, you _could_ think of a "parallel" flag, which when enabled would turn it on, otherwise it would be serial. However, how would you implement this? You would still need a variable with a default (`SERIAL`) value and a value to change it. So in the end, you end up exactly with what we have now. I am not sure how you think a "flag" should be implemented here.

Reply:
Yeah with flag I didn't mean in exclusively in the "user interface" way but as in a boolean. So it would work the same way basically no? I guess the relevant change to understand what I mean is in `aiida/schedulers/scheduler.py`, going from this:

``` python
        if codes_run_mode == CodeRunMode.PARALLEL:
            list_of_runlines.append('wait\n')
            return ' &\n\n'.join(list_of_runlines)

        if codes_run_mode == CodeRunMode.SERIAL:
            return '\n\n'.join(list_of_runlines)
```

To this:

``` python
        if do_parallel_execution:
            list_of_runlines.append('wait\n')
            return ' &\n\n'.join(list_of_runlines)
        else:
            return '\n\n'.join(list_of_runlines)
```

It is a super minor change, but I was just wondering what was the advantage of having an extra construct (however simple it may be) to add a layer of abstraction what seems to be a simple boolean switch.

---

Actually, nevermind: just by looking into this I changed my mind. By grepping `CodeRunMode` I was able to quickly identify in which parts the `run_mode` was being set, modified, and/or applied. That alone is a pretty good advantage in my books, so now I like this better haha. Sorry for having gone on a "circular tangent" 😅  (however contradictory that might be).